src/firefly/js/visualize/task/PlotAdminTask.js

```diff
@@ -86,23 +88,53 @@ export function changePointSelectionActionCreator(rawAction) {
     };
 }
 
+
 /**
  * @param {Action} rawAction
  * @returns {Function}
  */
 export function restoreDefaultsActionCreator(rawAction) {
     return (dispatcher, getState) => {
-        const vr= getState()[IMAGE_PLOT_KEY];
-        const {plotId}= rawAction.payload;
-        const {plotViewAry, positionLock}= vr;
-        const pv= getPlotViewById(vr,plotId);
+        const vr = getState()[IMAGE_PLOT_KEY];
+        const {plotId} = rawAction.payload;
+        const {plotViewAry, positionLock} = vr;
+        const pv = getPlotViewById(vr, plotId);
+
+
+        const changeColorsToDefault= (vr, hasOverlayColorLock) =>
+            vr.plotViewAry
+                .map( (pv)=>  { return primePlot(pv);})
+                .filter( (plot)=> {
+                     if (hasOverlayColorLock ) { //change for all images
+                         if (isImage(plot) && plot.plotState.getColorTableId() !== plot.plotState.getWebPlotRequest().getInitialColorTable()) {
+                             dispatchColorChange({
+                                 plotId: plot.plotId, cbarId: plot.plotState.getWebPlotRequest().getInitialColorTable(),
+                                 actionScope: ActionScope.SINGLE
+                             });
+                         }
+                     }
+                     else { //change only for the active image
+                         if (plot.plotId===vr.activePlotId && isImage(plot) && plot.plotState.getColorTableId() !== plot.plotState.getWebPlotRequest().getInitialColorTable()){
+                             dispatchColorChange({
+                                 plotId: plot.plotId, cbarId: plot.plotState.getWebPlotRequest().getInitialColorTable(),
+                                 actionScope: ActionScope.SINGLE
+                             });
+
+                         }
+                     }
+                }
+        );
```

Comment:
You don't need to do anything if hasOverlayColorLock is false (i.e. not `else`) because because line 133 will take care of the single case.

Reply:
Yes, you are right on the line 133.